src/sig_stfl/sig_stfl.h

```diff
@@ -0,0 +1,578 @@
+	 *
+	 * @param[out] public_key The public key is represented as a byte string.
+	 * @param[out] secret_key The secret key object
+	 * @return OQS_SUCCESS or OQS_ERROR
+	 */
+	OQS_STATUS (*keypair)(uint8_t *public_key, OQS_SIG_STFL_SECRET_KEY *secret_key);
+
+	/**
+	 * Signature generation algorithm.
+	 *
+	 * Caller is responsible for allocating sufficient memory for `signature`,
+	 * based on the `length_*` members in this object or the per-scheme
+	 * compile-time macros `OQS_SIG_STFL_*_length_*`.
+	 *
+	 * @param[out] signature The signature on the message is represented as a byte string.
+	 * @param[out] signature_len The length of the signature.
+	 * @param[in] message The message to sign is represented as a byte string.
+	 * @param[in] message_len The length of the message to sign.
+	 * @param[in] secret_key The secret key object pointer.
+	 * @return OQS_SUCCESS or OQS_ERROR
+	 */
+	OQS_STATUS (*sign)(uint8_t *signature, size_t *signature_len, const uint8_t *message, size_t message_len, OQS_SIG_STFL_SECRET_KEY *secret_key);
+
+	/**
+	 * Signature verification algorithm.
+	 *
+	 * @param[in] message The message is represented as a byte string.
+	 * @param[in] message_len The length of the message.
+	 * @param[in] signature The signature on the message is represented as a byte string.
+	 * @param[in] signature_len The length of the signature.
+	 * @param[in] public_key The public key is represented as a byte string.
+	 * @return OQS_SUCCESS or OQS_ERROR
+	 */
+	OQS_STATUS (*verify)(const uint8_t *message, size_t message_len, const uint8_t *signature, size_t signature_len, const uint8_t *public_key);
+
+	/**
+	 * Query number of remaining signatures
+	 *
+	 * @param[out] remain The number of remaining signatures
+	 * @param[in] secret_key The secret key object pointer.
+	 * @return OQS_SUCCESS or OQS_ERROR
+	 */
+	OQS_STATUS (*sigs_remaining)(unsigned long long *remain, const OQS_SIG_STFL_SECRET_KEY *secret_key);
+
+	/**
+	 * Total number of signatures
+	 *
+	 * @param[out] total The total number of signatures
+	 * @param[in] secret_key The secret key key object pointer.
+	 * @return OQS_SUCCESS or OQS_ERROR
+	 */
+	OQS_STATUS (*sigs_total)(unsigned long long *total, const OQS_SIG_STFL_SECRET_KEY *secret_key);
+
+} OQS_SIG_STFL;
+
+/**
+ * @brief OQS_SIG_STFL_SECRET_KEY object for stateful signature schemes
+ */
+
+typedef struct OQS_SIG_STFL_SECRET_KEY {
+
+	/** Associated signature object */
+	OQS_SIG_STFL *sig;
+
+	/* The (maximum) length, in bytes, of secret keys for this signature scheme. */
+	size_t length_secret_key;
+
+	/* The variant-specific secret key data, must be allocated at the initialization. */
+	void *secret_key_data;
+
+	/* mutual exclusion struct */
+	void *mutex;
+
+	/* Application managed data related to secure storage of secret key data */
+	void *context;
+
+	/**
+	 * Secret Key retrieval Function
+	 *
+	 * @param[in] sk The secret key represented as OQS_SIG_STFL_SECRET_KEY object
```

Comment:
suggest documentation to order as per actual parameter ordering

Reply:
Fixed in https://github.com/open-quantum-safe/liboqs/pull/1655